middleware/faq.rst

```diff
@@ -535,6 +535,251 @@ The resulting query logs can be useful for developers and database administrator
 
 .. _middleware_faq_clean_up_runs:
 
+How can I get a report on all failures and missing datasets in a run?
+=====================================================================
+.. _middleware_faq_pipetask_report:
+
+The :any:`pipetask report <lsst.ctrl.mpexec-pipetask#report>` tool can be used to analyze executed quantum graphs, troubleshoot, diagnose failures, and confirm that fail-and-recovery attempts (such as when using ``--skip-existing-in``) are effective.
+
+When analyzing multiple graphs with ``pipetask report``, all graphs should be attempts to execute the same pipeline with the same dataquery.
+
+The recommended usage is
+
+.. code:: sh
+
+  pipetask report --full-output-filename <path/to/output_file>.json --force-v2 REPO QGRAPHS
+
+
+- The ``<path/to/output_file>.json`` option is the path to a file where the output can be stored
+- The ``--force-v2`` option makes sure that the most recent version of the tool is used even when the user passes only one graph
+- The ``REPO`` argument is the `Butler` repo where the output from the processing is stored
+- The ``QGRAPHS`` argument is a ``Sequence`` of `QuantumGraph`s to be analyzed, separated by spaces and passed in order of first to last executed
+
+This will print two ``bps report``-style tables, one for quanta and one for output datasets.
+
+In the output JSON file will be
+
+- A summary under every task with: 
+
+  * Every failed data ID and corresponding error message
+  * Every run containing failing data IDs, and their status
+  * A list of data IDs which have been "recovered"; i.e., successes from fail-and-recovery attempts
+- A list of the data IDs associated with every missing dataset
+- A field called ``producer`` connecting each ``datasetType`` for each data ID to the task which produced it
+- Counts of quanta and datasets in all possible states. 
+
+Currently, the ``--force-v2`` option is the suggested usage until version 1 of pipetask report (using the `QuantumGraphExecutionReport` instead of the `QuantumProvenanceGraph`) is deprecated.
+
+With that, we'll go into how to read the output of ``pipetask report``.
+
+The Quanta Table
+----------------
+
+The table for Quanta from the ``w_2024_28`` DC2 test-med-1 reprocessing run, before recoveries, looked like:
+
+.. code:: sh
+
+               Task          Unknown Successful Blocked Failed Wonky TOTAL EXPECTED
+    ---------------------- ------- ---------- ------- ------ ----- ----- --------
```

Comment:
It looks like the indentation for some of the column headers of this table became misaligned.